cmd/create/machinepool/machinepool_test.go

```diff
@@ -0,0 +1,43 @@
+package machinepool
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("MachinePool", func() {
+	DescribeTable("ParseTaints validation",
+		func(taint, expectedError string, numberOfTaints int) {
+			taints, err := parseTaints(taint)
+			if expectedError == "" {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(taints)).To(Equal(numberOfTaints))
+			} else {
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(expectedError))
+			}
+		},
+		Entry(
+			"Well formed taint",
+			"node-role.kubernetes.io/infra=val:NoSchedule", "", 1),
+		Entry(
+			"2 well formed taints",
+			"node-role.kubernetes.io/infra=val:NoSchedule,node-role.kubernetes.io/master=val:NoSchedule", "", 2),
+		Entry(
+			"Empty value taint",
+			"node-role.kubernetes.io/infraNoSchedule",
+			"Expected key=value:scheduleType format", 0),
+		Entry(
+			"Empty value taint",
+			"node-role.kubernetes.io/infra=:NoSchedule",
+			"Expected key=value:scheduleType format", 0),
+		Entry(
+			"Empty value taint",
+			"node-role.kubernetes.io/infra=val:NoSchedule,node-role.kubernetes.io/infra=:NoSchedule",
+			"Expected key=value:scheduleType format", 0),
+		Entry(
+			"Empty effect taint",
+			"node-role.kubernetes.io/infra=:",
+			"Expected key=value:scheduleType format", 0),
+	)
```

Comment:
what about `node-role.kubernetes.io/infra=foobar:NoSchedule`? is that a valid taint?

Reply:
yes. This is what I am testing in https://github.com/openshift/rosa/pull/1125/files#diff-2c663107801b3bd651998b1d576dbf34aa8d2826e2078c57a865c22775183707R22 if I am not missing something

Reply:
https://issues.redhat.com/browse/OPCT-13

Related btw

Reply:
interesting @gdbranco . This one is just to avoid the panic but indeed we need to allow empty values both here and in the CS. Should we do it separately or you want to fix the 2 issues together?
@tbrisker val is just a random string, I will add another one for remove any confusion